### DISCUSSION TITLE
B1 scaling factor

### DISCUSSION THREAD

**Discussion by schote** [schote/nexus-console#67] [Ideas]
The RF amplitude in pulseq is defined in Hz. 
When unrolling the sequence, the amplitude in Hz must be translated to a voltage in mV.

We currently have two scaling factors for the translation.
A coarse scaling factor defined in `device_config.yaml` which brings the amplitude close to 1 and a finer scaling factor which is adjusted to the object (usually it is in the range between 1 and 10).
However, the coarse scaling value was chosen empirically and the second finer scaling factor is optimized to the object.

It would be nice to have a comprehensible value which depends on the maximum output that can be achieved with a setup, so that the maximum B1-scaling value corresponds to 1.

**Comment by schote** [comment]
From the device parameter perspective the flip angle is limited by the maximum channel output level. To have a more meaningful range  for the scaling factor, we would need to agree on a maximum amplitude in Hz, i.e. a maximum flip angle?
Is there any reasonable maximum value to agree on, @LUMC-LowFieldMRI ?

**Comment by LUMC-LowFieldMRI** [comment]
I recently implemented something similar on the python version of the Kea console which is a B1 efficiency in B1 per Volt, with the voltage being the output of the RF channel of the console. If you have that it's trivial to calculate the voltage needed for any flip angle/pulse duration (assuming linearity). This is probably the way to do it here too, even if the scaling to output voltage is not immediately obvious it doesnt matter, as long as it scales consistently.

This can be expanded if the gain of the RF amplifier is known, if this is a variable that can be given we know the power going in to the coil which is relevant for SAR.

**Reply by schote** [reply]
Do you mean the output of the RF channel in terms of the maximum?

**Reply by LUMC-LowFieldMRI** [reply]
No, voltage in the absolute sense

**Reply by schote** [reply]
How did you measure the B1 efficiency for your implementation?
